Travis CI: Bind remaining python modules

cmonr opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->

Fixes https://github.com/ARMmbed/mbed-os/issues/9381

After this PR, all python modules used in CI _should_ be loosely version-bound.

We can get stricter from here, but this _should_ a majority of issues like [this](https://github.com/ARMmbed/mbed-os/pull/9371) and [this](https://github.com/ARMmbed/mbed-os/pull/9131) from happening again.

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!-- 
    Optional
    Request additional reviewers with @username
-->

bridadan commented on requirements.txt:
```diff
@@ -20,3 +20,4 @@ six==1.11.0
 git+https://github.com/armmbed/manifest-tool.git@v1.4.6
 mbed-cloud-sdk==2.0.1
 icetea>=1.0.2,<1.1
+idna==2.7 # Required by `icetea`
```
I would really discourage us from doing things like this, especially if we don't actually depend on it in the tools contained within this repository. Is this requirement not already present in icetea?

cmonr replied:
I didn't want to do this either, but as it is, Mbed OS is a lot more stricter with versions than icetea:
- https://github.com/ARMmbed/icetea/blob/master/requirements.txt
- https://github.com/ARMmbed/icetea/blob/master/dev_requirements.txt
- https://github.com/ARMmbed/icetea/blob/master/extra_requirements.txt

ARMmbed/icetea uses modules on the bleeding edge, wheras Mbed OS does not. The reason Mbed OS is squeaking along (imo) is that we cache our pip env in CI.

bridadan replied:
I see. Hmm. Have we ever mentioned to the icetea to at least pin their major versions? (I realize you're just the messenger here, sorry for pestering).

cmonr replied:
I hadn't, but have now: https://github.com/ARMmbed/icetea/issues/69

Priority was to get CI stable in Mbed OS first, as it seems that this is now affecting the release build...

cmonr commented:
As the new commits show, I'm still looking into this.

It didn't help that mbedmicro/pyOCD just published v0.15, which has incompatabilities with pyYaml due to the security advisory being fixed here.

bridadan reviewed (commented):
I'm lifting my red X because I'm a little more familiar with the issue. Mbed OS doesn't have much options in this case. I'll be making a release of greentea and htrun that will make the pyocd dependency optional, so hopefully that will help with some of the issues.

Commit:
Added specific versions of Travis-only modules
Use requirements.txt to install specific versions of Travis CI python modules.

Commit:
Shuffled around version checks to be more useful in case of failure

Commit:
Added warning suppression to python api_tests.
Apparently local laptop isn't fast enough for test generation, but CI is

Commit:
Updated requirements.txt to be more module-version resilliant.
Moved modules tied to specific versions higher, and modules with no restrictions lower.

cmonr commented:
Ready for a re-review.

cmonr commented:
**NOTE:** Holding until tools are updated to make pyocd optional.

Otherwise the following line needs to be added into `requirements.txt`:
`pyocd>=0.14,<0.15`

@bridadan Will be doing that tomorrow-ish.

Commit:
Added pyocd requirement.
Dependency tree: icetea > mbed_flasher > pyocd

cmonr commented:
@ARMmbed/mbed-os-maintainers One review away from being able to start CI, and have considerably more defined reqs!

0xc0170 commented:
CI started
